corehq/apps/locations/models.py

```diff
@@ -549,6 +550,43 @@ def _remove_user(self):
             user.active = False
             user.save()
 
+    def deprecate_by(self, other_location, operation, archive=True):
+        """
+        add metadata on locations
+        for this/self location
+        1. deprecated_to: this can be a single location or a list like in case of a split
+        2. deprecation_operation: this location was deprecated by a merge/split/extract/move operation
+        3. deprecated_at: In case of split, since multiple locations will deprecate it, this would be a dict
+        with new location id mapped to the timestamp of deprecation performed
+        for other location
+        1. deprecates: note this location id on it. This can hold multiple values like in case of a merge a new
+        location would come from multiple locations
+        :param other_location: the location that deprecates this location
+        :param operation: the operation being performed, split/merge/move/extract
+        :param archive: to archive this location or not
+        :return: True for success, False in case of failure
+        """
+        other_location_id = other_location.location_id
+        if 'deprecated_to' in self.metadata:
```

Comment:
What would you think of making these keys constants?

Reply:
https://github.com/dimagi/commcare-hq/pull/26898/commits/897e3a6be6ff820317bc5e6d9fa1c2fb0f4707c8